src/ui/DynamicHierarchicalFacet/DynamicHierarchicalFacet.ts

```diff
@@ -206,6 +207,19 @@ export class DynamicHierarchicalFacet extends Component implements IDynamicHiera
      * **Default:** `true` if folded results are requested;`false` otherwise.
      */
     filterFacetCount: ComponentOptions.buildBooleanOption({ section: 'Filtering' }),
+
+    /**
+     * The sort criterion to use for this facet.
+     *
+     * Allowed values are `occurences` and `alphanumeric`.
+     *
+     * **Default (API):** `occurrences`
+     */
+    sortCriteria: <FacetSortCriteria>ComponentOptions.buildStringOption({
```

Comment:
Similar comment as above on leveraging the generic argument of `buildStringOption`. However, we should not use `FacetSortCriteria` because the type is not constrained enough. Someone using intellisense in the IEditor will see all options available instead of just two. Would the following work?

```
ComponentOptions.buildStringOption<FacetSortCriteria.alphanumeric | FacetSortCriteria.occurences>(...)
```

Alternatively, we could store that in a type to be more concise or if it's going to be reused in other places:
```
type HierarchicalFacetSortCriteria = FacetSortCriteria.alphanumeric | FacetSortCriteria.occurrences;

ComponentOptions.buildStringOption<HierarchicalFacetSortCriteria>(...)
```